services/streammanager/kafka/kafkamanager.go

```diff
@@ -394,7 +419,38 @@ func prepareMessage(topic, key string, message []byte, timestamp time.Time) clie
 	}
 }
 
-func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timestamp time.Time) (
+// This function is used to create the error string of an individual schema
+func createErrString(err error, index int) string {
+	return fmt.Sprintf(" Schema[%d]: ", index+1) + err.Error() + "."
+}
+
+// This function is used to serialize the binary data according to the avroSchema.
+// It iterates over the schemas provided by the customer and tries to serialize the data.
+// If it's able to serialize the data then it returns the converted data otherwise it returns an error.
+// We are using the LinkedIn goavro library for data serialization. Ref: https://github.com/linkedin/goavro
+func serialize(value []byte, codecs []goavro.Codec) ([]byte, error) {
+	var errorStr string
+	for i, codec := range codecs {
```

Comment:
as @fracasula raised this may end up to some serious performance degradation ...

Reply:
Removed it and added support of schemaId which will be available with every message and based on the schemaId we will use the codec.